src/kontrol/options.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from collections.abc import Iterable
+
+    from pyk.utils import BugReport
+
+
+@dataclass(frozen=True)
+class ProveOptions:
+    auto_abstract_gas: bool
+    bug_report: BugReport | None
+    use_booster: bool
+    kore_rpc_command: str | Iterable[str] | None
+    smt_timeout: int | None
+    smt_retry_limit: int | None
+    trace_rewrites: bool
+    simplify_init: bool
+    bmc_depth: int | None
+    max_depth: int
+    break_every_step: bool
+    break_on_jumpi: bool
+    break_on_calls: bool
+    workers: int
+    counterexample_info: bool
+    max_iterations: int | None
+    run_constructor: bool
+    fail_fast: bool
+    reinit: bool
+    port: int | None
+
+    def __init__(
+        self,
+        auto_abstract_gas: bool = False,
+        bug_report: BugReport | None = None,
+        use_booster: bool = True,
+        kore_rpc_command: str | Iterable[str] | None = None,
+        smt_timeout: int | None = None,
+        smt_retry_limit: int | None = None,
+        trace_rewrites: bool = False,
+        bmc_depth: int | None = None,
+        max_depth: int = 1000,
+        break_every_step: bool = False,
+        break_on_jumpi: bool = False,
+        break_on_calls: bool = True,
+        workers: int = 1,
+        counterexample_info: bool = False,
+        max_iterations: int | None = None,
+        run_constructor: bool = False,
+        fail_fast: bool = True,
+        reinit: bool = False,
+        port: int | None = None,
+    ) -> None:
+        if kore_rpc_command is None:
+            kore_rpc_command = ('kore-rpc-booster',) if use_booster else ('kore-rpc',)
```

Comment:
To ensure declared types are as straightforward as possible, you can for example:

```suggestion
        if kore_rpc_command is None:
            kore_rpc_command = ('kore-rpc-booster',) if use_booster else ('kore-rpc',)
        elif isinstance(kore_rpc_command, str):
            kore_rpc_command = (kore_rpc_command,)
        else:
            kore_rpc_command = tuple(kore_rpc_command)
```

Reply:
I added this. For the other fields, I left the types as-is since for the ones that can be `None` this has a specific meaning in the rest of the code.